pyproject.toml

```diff
@@ -75,7 +87,7 @@ before-all = [
 select = "*-win_amd64"
 
 [tool.cibuildwheel.windows]
-test-command = '''bash -c "cd '{project}/bindings/python' && tox -e"'''
+test-command = '''bash -c "cd '{project}/bindings/python' && python -m pip install -r test-requirements.txt && python -X dev -m pytest --log-level=DEBUG --log-cli-level=DEBUG && python -m mypy --config-file mypy-tox.ini tests"'''
```

Comment:
once installing the test dependencies, would it not work to just run `pytest tests`?

Reply:
This copies the testing invocation from `tox`. It's not sufficient to invoke `tox` because it creates an isolated environment, and we want to test in the outer `cibuildwheel` environment. I'll factor these invocations out somewhere so it can be shared.

This config block is only duplicated in `pyproject.toml` several times due to shell quoting issues on windows. If I factor it out into a script, that can probably be avoided.

I wrote some justification for why we test what we test in `tox.ini`. The highlights are that `python -X dev` has been super useful in the past for catching bugs and producing good stack traces, and that this particular `mypy` run is crucial for testing that the type stubs got *installed* correctly from the build artifact. Running `mypy` in the source tree just tests the type stubs in the tree.